git-remote-ipfs/src/Network/IPFS/Git/RemoteHelper.hs

```diff
@@ -59,10 +64,22 @@ renderProcessError = \case
     CidError  e       -> fmt ("Cid conversion error: " % fstr) e
     UnknownLocalRef r -> fmt ("Ref not found locally: " % ftxt) r
 
-processCommand
-    :: (MonadMask m, MonadIO m)
-    => Command
-    -> RemoteHelperT ProcessError m CommandResult
+newtype ConcurrentException =
+    ConcurrentException (RemoteHelperError ProcessError)
+
+instance Show ConcurrentException where
+    show (ConcurrentException e) =
+        sfmt ( "Network.IPFS.Git.RemoteHelper.ConcurrentException: "
```

Comment:
I think my previous comment got swallowed by the UI, but was saying that if you don't want to implement a `Show` instance, but you still want the exception to be displayed, you can always try implementing `displayException` explicitly and see if that works out.